docs/DebuggerHTML-Components-Reducers-Actions.md

```diff
@@ -0,0 +1,663 @@
+<pre>javascript
+const reducers = require("./reducers");
+
+.
+
+.
+
+const store = createStore(combineReducers(reducers));</pre>
+
+All of the reducers are combined using the index.js file in the
+reducers folder. In the Debbuger.html project, each reducer has an
+<code>update()</code> function to handle actions for its slice of state.
+
+##Asynch-requests
+
+The **asynch-requests** reducer creates an array that stores a unique
+sequence number for every promise being executed from an action.
+It removes the sequence number from the array when a specific promise
+resolves or rejects. The following image shows a snapshot of the
+debugger.html state with an active promise.
+
+![](images/asynchreducer.png)
+
+##Breakpoints
+
+The **breakpoints reducer** is responsible for handling breakpoint
+state. It adds an Immutable Map of breakpoint objects and a Boolean
+value for whether breakpoints are disabled to the application state.
+
+![](images/breakpoints1.png)
+
+Each breakpoint object in the map contains information like the
+location, the actual source file, whether the breakpoint is disabled,
+a unique id, and the text for the breakpoint. The following is an example
+of what the breakpoints state looks like for a selected breakpoint:
+
+![](images/breakpoints2.png)
+
+The breakpoints reducer handles several action types. The actions
+handled by this reducer are all fired wrapped in a promise. The status
+of the promise can be checked in the action object using code similar to
+the following:
+
+<pre>javascript
+if (action.status === "start") {</pre>
+
+Valid values are <code>start</code>, <code>done</code> and <code>error</code>.
+
+The following action types are handled:
+
+-   <code>ADD\_BREAKPOINT</code> - This command adds breakpoints to the state as
+    shown in the image above. While the promise is being fulfilled the
+    loading attribute is set to <code>true</code> and a basic breakpoint object
+    is created. If the promise is completed, the location is updated
+    and the loading attribute is set to <code>false</code>. If the promise fails, the
+    breakpoint is deleted from the state.
+
+-   <code>TOGGLE\_BREAKPOINTS</code> - Sets the value of the <code>breakpointsDisabled</code>
+    attribute and is used to disable or enable all breakpoints.
+
+-   <code>SET\_BREAKPOINT\_CONDITION</code> - This action sets the condition
+    attribute for a specific breakpoint. This functionality is currently not implemented in the UI.
+
+-   <code>REMOVE\_BREAKPOINT</code> - This action actually handles disabling or
+    removing a breakpoint. If the breakpoint is being disabled the
+    disabled attribute of the breakpoint is set to <code>false</code>. If
+    the breakpoint is removed it is deleted from the breakpoints state.
+    
+
+The breakpoints reducer additionally supplies functions to
+retrieve information from the breakpoints state. For example the
+<code>getBreakpointsForSource()</code> function returns all breakpoints for a given

```

Comment:
> We can get this landed as is and then separate out the reducer

> I think docs/reducers/ might be better so we keep it all accessible from the github pages

@jlongster what was the documentation tool you were using earlier? do you think it could pull from directory level doc directories?